### ISSUE TITLE
Using asinf may leed to nan (not a number)

### ISSUE THREAD

**Issue by the-m-master** [xioTechnologies/Fusion#25]
Using 'asinf()' may lead to nan. In the function 'FusionQuaternionToEuler' a call is done to 'asinf', if the input of this is larger then 1 or less then -1, the result will be nan. All calculations using this pitch value (as a nan) will fail as well.

A potential solution might be something like:

float asinf_safe(float val)  {
  if (val >= 1.0f) { return (M_PI / 2.0f); }
  if (val <= -1.0f) { return (M_PI / -2.0f); }
  return asinf(val);
}

**Comment by xioTechnologies** [comment]
Thank you for raising the issue.  The problem can be demonstrated using the following code.  The plot ends after 45 samples because the remaining 55 samples are all NaN.

```python
import imufusion
import matplotlib.pyplot as pyplot
import numpy

numberOfSamples = 100

ahrs = imufusion.Ahrs()
pitch = numpy.empty(numberOfSamples)

for index in range(numberOfSamples):
    ahrs.update_no_magnetometer(numpy.array([0, 0, 0]), numpy.array([1, 0, 0]), 1 / numberOfSamples)
    pitch[index] = ahrs.quaternion.to_euler()[1]

pyplot.plot(pitch)
pyplot.show()
```
![image](https://user-images.githubusercontent.com/1748389/172851367-3742a0c4-7e60-4ba4-bbc5-3e23d4c2a070.png)

After implementing the fix you have suggested, the plot shows all 100 samples as shown below.

![image](https://user-images.githubusercontent.com/1748389/172851548-dcb04e99-032a-4670-a32c-42ccf9f48eb4.png)

As an aside, it is interesting to note that the transition from the final `asinf` result (-89.5) to the hardcoded `-90.0f` is visible in the above plot as step change.  The transition becomes smoother when `FUSION_USE_NORMAL_SQRT` is used as shown below.

![image](https://user-images.githubusercontent.com/1748389/172851853-46c07551-9c8f-4c60-b58c-4c6e59586986.png)

**Comment by xioTechnologies** [comment]
The fix  has been applied as commit [d7788de](https://github.com/xioTechnologies/Fusion/commit/d7788de2114c544d87cba22bd9802dd9244b1d9e).

**Comment by the-m-master** [comment]
Thanks! We are going to use it in the next release :)
